// @bridged.xyz/client-sdk/dist/g11n/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalizedVariable = exports.GlobalizedKey = exports.Locale = void 0;
__exportStar(require("./validators"), exports);
__exportStar(require("./placeholder"), exports);
class Locale {
    constructor(props) {
        this.languageCode = props.languageCode;
        this.countryCode = props.countryCode;
    }
    toString() {
        return `${this.languageCode}${this.countryCode ? `_${this.countryCode}` : ''}`;
    }
}
exports.Locale = Locale;
class GlobalizedKey {
    constructor(props) {
        this.id = props.id;
        this.key = props.key;
        this.translations = props.translations;
    }
    updateValueFor(locale, newValue) {
    }
}
exports.GlobalizedKey = GlobalizedKey;
class GlobalizedVariable {
    constructor(props) {
        this.id = props.id;
        this.name = props.name;
        this.namespace = props.namespace;
        this.translations = props.translations;
    }
}
exports.GlobalizedVariable = GlobalizedVariable;
