// @bridged.xyz/client-sdk/dist/build/flutter/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildAndHostSimpleApp = exports.compileFlutterApp = void 0;
const dart_services_1 = require("dart-services");
const url_1 = require("../../url");
const hosting_1 = require("../../hosting");
async function compileFlutterApp(props) {
    const compiled = await dart_services_1.compileComplete(props.dart);
    if (!compiled.sucess) {
        throw `compile failed with error ${JSON.stringify(compiled.error, null, 2)}`;
    }
    return {
        id: props.id,
        js: compiled.result
    };
}
exports.compileFlutterApp = compileFlutterApp;
async function buildAndHostSimpleApp(props) {
    const compiled = await compileFlutterApp({
        dart: props.dart,
        id: props.id
    });
    const hosted = await hosting_1.upload({
        file: compiled.js,
        name: props.id
    });
    if (props.short) {
        const short = await url_1.makeShortUrl({
            url: hosted.url
        });
        return short.url;
    }
    else {
        return hosted.url;
    }
}
exports.buildAndHostSimpleApp = buildAndHostSimpleApp;
