// @bridged.xyz/client-sdk/dist/g11n/repository/layer-key-map.repository.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LayerKeyMapRepository = void 0;
const api_1 = require("../api");
class LayerKeyMapRepository {
    constructor(projectId, sceneId) {
        this.projectId = projectId;
        this.sceneId = sceneId;
        this.layerKeyMap = new Map();
    }
    async putMap(layerId, keyId) {
        this.layerKeyMap.set(layerId, keyId);
        await api_1.putLayerKeyMap(this.projectId, {
            layerId: layerId,
            keyId: keyId,
            sceneId: this.sceneId
        });
    }
    async fetchLayerTranslation(layerId) {
        const t = await api_1.fetchTextTranslationFromLayer(this.projectId, {
            layerId: layerId,
            sceneId: this.sceneId
        });
        return t;
    }
    async fetchTranslations() {
        return await api_1.fetchTextTranslationsFromScene(this.projectId, {
            sceneId: this.sceneId
        });
    }
}
exports.LayerKeyMapRepository = LayerKeyMapRepository;
