// @bridged.xyz/client-sdk/dist/assets/api/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateVariant = exports.addAvariant = exports.putVariant = exports.getVariantAsset = exports.registerVariantAsset = void 0;
const axios_1 = __importDefault(require("axios"));
const hosts_1 = require("../../constants/hosts");
const VARIANT_ASSETS_ROUTE = "variant-assets";
const axios = axios_1.default.create({
    baseURL: hosts_1.ASSET_SERVICE_HOST
});
async function registerVariantAsset(projectId, req) {
    const resp = await axios.post(VARIANT_ASSETS_ROUTE, req);
    return resp.data;
}
exports.registerVariantAsset = registerVariantAsset;
async function getVariantAsset(projectId, id) {
    const resp = await axios.get(`${VARIANT_ASSETS_ROUTE}/${id}`);
    return resp.data;
}
exports.getVariantAsset = getVariantAsset;
async function putVariant(projectId, request) {
    try {
        const resp = await axios.put(`${VARIANT_ASSETS_ROUTE}/${request.variantAssetId}/variants/${request.variant}`, request.asset);
        return resp.data;
    }
    catch (_) {
        console.error('an unkwon error occured while performing putVariant', _);
    }
}
exports.putVariant = putVariant;
async function addAvariant(projectId, request) {
    const resp = await axios.post(`${VARIANT_ASSETS_ROUTE}/${request.variantAssetId}/variants/${request.variant}`, request.asset);
    return resp.data;
}
exports.addAvariant = addAvariant;
async function updateVariant(projectId, request) {
    const resp = await axios.patch(`${VARIANT_ASSETS_ROUTE}/${request.variantAssetId}/variants/${request.variant}`, request.asset);
    return resp.data;
}
exports.updateVariant = updateVariant;
